public/services/shared-link.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { parse } from 'url';
+import { CoreStart } from 'kibana/public';
+import { API_ENDPOINT_MULTITENANCY } from '../apps/configuration/constants';
+
+// refer to https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/master/public/chrome/multitenancy/enable_multitenancy.js
+export async function addTenantToShareURL(core: CoreStart) {
+  let tenant = '';
+  try {
+    tenant = await core.http.get(API_ENDPOINT_MULTITENANCY);
```

Comment:
Will this run before login? Do we need to ignore 403?

Reply:
yes it execute on login page, we log the error in console and do nothing here, like what we did for permissionInfo call